client/src/components/common/SearchBar.tsx

```diff
@@ -0,0 +1,82 @@
+"use client";
```

Comment:
u need to use hooks to break this down, 1 hook to be the api layer and another hook to be the go between the api layer and this component